sms_statistics_service/src/firestore_wrapper.py

```diff
@@ -47,14 +62,23 @@ def update_sms_stats(self, project_name, sms_stats_batch):
         :type sms_stats_batch: dict of str -> src.data_models.SMSStats
         """
         log.info(f"Batch updating {len(sms_stats_batch)} SMS stats for project {project_name}...")
+        batch = self.client.batch()
+        batch_counter = 0
+        total_progress = 0
+        for iso_string, sms_stats in sms_stats_batch.items():
+            batch.set(self._get_sms_stat_doc_ref(project_name, iso_string), sms_stats.to_dict())
+            batch_counter += 1
+            total_progress += 1
 
-        doc_ref = self._get_sms_stats_doc_ref(project_name)
-        updates = {iso_string: sms_stats.to_dict() for iso_string, sms_stats in sms_stats_batch.items()}
+            if batch_counter >= self.MAX_BATCH_SIZE:
+                batch.commit()
+                log.info(f"Batch of {batch_counter} messages committed, progress: {total_progress} / {len(sms_stats_batch)}")
+                batch = self.client.batch()
+                batch_counter = 0
 
-        if not doc_ref.get().exists:
-            log.warning(f"No SMS stats document exists for project {project_name}; will create one...")
-            doc_ref.set(updates)
-        else:
-            doc_ref.update(updates)
+        if batch_counter > 0:
+            batch.commit()
+            log.info(f"Final batch of {batch_counter} messages committed")
```

Comment:
I usually reset the counter here just out of good house-keeping, and also that we can assert that the counter is zero at the end

Reply:
done